api/v1alpha3/conversion.go

```diff
@@ -96,13 +98,43 @@ func (dst *MachineSetList) ConvertFrom(srcRaw conversion.Hub) error {
 func (src *MachineDeployment) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha4.MachineDeployment)
 
-	return  Convert_v1alpha3_MachineDeployment_To_v1alpha4_MachineDeployment(src, dst, nil)
+	if err := Convert_v1alpha3_MachineDeployment_To_v1alpha4_MachineDeployment(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &v1alpha4.MachineDeployment{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	if restored.Spec.Strategy != nil && restored.Spec.Strategy.RollingUpdate != nil {
```

Comment:
This doesn't really matter because v1alpha3 only supports the RollingUpdate strategy, but if we wanted to be "more correct", this logic should only execute if `dst.Spec.Strategy.Type == "RollingUpdate"`.

Reply:
This is making the fuzz tests fail because it tries to set delete policy even when `spec.strategy.Type` is not `RollingUpdate`. I think I'll need to modify the fuzzer to account for this logic. Should I create a separate issue to track the fuzz tests failure including the `spoke-hub-spoke` ones?